packages/components/src/systems/SignatureAttestation/SignatureAttestationModal.tsx

```diff
@@ -0,0 +1,252 @@
+  agreeToSignatureAttestation?: boolean;
+};
+
+type DataReturn<Type> = {
+  data?: Type;
+  loading?: boolean;
+  error?: any;
+  errors?: readonly any[];
+};
+
+const getCurrentUserSignatureAttestationStatus = async (
+  client: ApolloClient<any>
+): Promise<DataReturn<GetCurrentUserSignatureAttestationStatusType>> => {
+  return await client.query<GetCurrentUserSignatureAttestationStatusType>({
+    query: GetCurrentUserSignatureAttestationStatus,
+    fetchPolicy: 'cache-first'
+  });
+};
+
+const agreeToSignatureAttestation =
+  (client: ApolloClient<any>) =>
+  async (version: string): Promise<DataReturn<Maybe<AgreeToSignatureAttestationType>>> => {
+    return await client.mutate<AgreeToSignatureAttestationType, { version: string }>({
+      mutation: AgreeToSignatureAttestation,
+      variables: { version },
+      refetchQueries: ['GetCurrentUserSignatureAttestationStatus'],
+      awaitRefetchQueries: true,
+      update: (cache, { data }) => {
+        if (data?.agreeToSignatureAttestation) {
+          cache.writeQuery({
+            query: GetCurrentUserSignatureAttestationStatus,
+            data: {
+              me: {
+                signatureAttestationStatus: {
+                  __typename: 'CompletedSignatureAttestation',
+                  agreedAt: new Date(),
+                  version
+                }
+              }
+            }
+          });
+        }
+      }
+    });
+  };
+
+export interface SignatureAttestationModalProps {
+  client: ApolloClient<any>;
+  children: JSXElement;
+  onAgreeComplete?: () => void;
+}
+
+type Status =
+  | {
+      status: 'LOADING';
+    }
+  | { status: 'NEEDS ATTESTATION'; version: string }
+  | { status: 'COMPLETE' }
+  | { status: 'ERROR'; errors: any[] };
+
+export const SignatureAttestationModal = (props: SignatureAttestationModalProps) => {
+  let ref: Ref<any> | undefined;
+  const [status, setStatus] = createSignal<Status>({ status: 'LOADING' });
+
+  const refreshAttestationStatus = async () => {
+    const req = await getCurrentUserSignatureAttestationStatus(props.client);
+    if (req.error || req.errors) {
+      setStatus({
+        status: 'ERROR',
+        errors: [
+          ...(req.errors?.map((e) => e.message) ?? []),
+          ...(req.error ? [req.error.message] : [])
+        ]
+      });
+    } else if (!req.data?.me?.signatureAttestationStatus) {
+      setStatus({
+        status: 'ERROR',
+        errors: ['Could not get user data. Please refresh and try again']
+      });
+    } else if (
+      req.data?.me?.signatureAttestationStatus?.__typename === 'CompletedSignatureAttestation' ||
+      req.data?.me?.signatureAttestationStatus?.__typename === 'NotApplicableSignatureAttestation'
+    ) {
+      setStatus({ status: 'COMPLETE' });
+    } else
+      setStatus({
+        status: 'NEEDS ATTESTATION',
+        version: req.data.me.signatureAttestationStatus.version
+      });
+  };
+
+  const dispatchSignatureAttestationAgreed = () => {
```

Comment:
you don't need to do this unless you're trying to communicate with something in the react app, otherwise the prescribe form internally is just solid so pass a callback function

Reply:
or I guess maybe you're trying to avoid drilling the callback through the prescribe wrapper to the prescribe element, all good in that case

Reply:
yeah exactly.